node/cli/src/chain_spec/tdx_testnet.rs

```
@@ -0,0 +1,97 @@
+// Copyright (C) 2023 Entropy Cryptography Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use crate::chain_spec::{dev::development_genesis_config, get_account_id_from_seed, ChainSpec};
+
+use entropy_runtime::wasm_binary_unwrap;
+use entropy_shared::{BoundedVecEncodedVerifyingKey, X25519PublicKey as TssX25519PublicKey};
+use sc_service::ChainType;
+use sp_core::sr25519;
+use sp_runtime::BoundedVec;
+
+/// The build time measurement value from the current entropy-tss VM images
+const ACCEPTED_MRTD: [u8; 48] = [
+    145, 235, 43, 68, 209, 65, 212, 236, 224, 159, 12, 117, 194, 197, 61, 36, 122, 60, 104, 237,
+    215, 250, 254, 138, 53, 32, 201, 66, 166, 4, 164, 7, 222, 3, 174, 109, 197, 248, 127, 39, 66,
+    139, 37, 56, 135, 49, 24, 183,
+];
+
+lazy_static::lazy_static! {
+    pub static ref PCK: BoundedVecEncodedVerifyingKey = vec![
+    2, 166, 103, 136, 58, 157, 155, 124, 186, 75, 81, 133, 87, 255, 233, 182, 192, 125, 235, 230,
+    121, 173, 147, 108, 47, 190, 240, 181, 75, 181, 31, 148, 128,
+    ].try_into().unwrap();
+}
+
+fn tdx_devnet_four_node_initial_tss_servers(
+) -> Vec<(sp_runtime::AccountId32, TssX25519PublicKey, String, BoundedVecEncodedVerifyingKey)> {
+    let tss_ip = std::env::var("ENTROPY_TESTNET_TSS_IP")
+        .expect("ENTROPY_TESTNET_TSS_IP environment variable to be set");
```

Why panic here instead of having a fallback to a default IP?

---

I'm not sure what default IP we can fallback to.  `localhost` wont work because these are the IPs which the TSS nodes use to communicate with each other, and from the CVM point of view localhost is not the same as localhost on the host.  I don't really want to hardcode the public IP of our current box.